# Medidas de tendencia central


## 1. ¿Qué son las medidas de tendencia central?

> Son una manera de cuantificar el valor más comúnmente encontrado en nuestros datos.

## 2. ¿Cuáles son las medidas de tendencia central más usadas?

- Datos numéricos:
    - La media (o promedio)
    - La mediana
- Datos categóricos:
    - La moda

### 2.1. La media (o promedio)

> Es simplemente la suma de todos los valores dividida entre el número total de valores **de la muestra**

Notación:
- $\bar{x}$ cuando nos referimos al promedio de la muestra
- $\mu$ cuando nos referimos al promedio de la población

Es una medida propia de la **distribución normal** así que se debería usar si nuestros datos numéricos tienen este tipo de distribución.

Ejemplo:

In [1]:
valores = [3, 5, 9, 8]
media = sum(valores)/len(valores)
print(f'Media = {media}')

Media = 6.25


La principal limitación de la media es que **es sensible a valores extremos** (***outliers***).

Por ejemplo, agreguemos un valor extremo a la lista anterior y veamos cómo afecta la media:

In [4]:
valores = [3, 5, 9, 8, 500]
media = sum(valores)/len(valores)
print(f'Media = {media}')

Media = 105.0


Vemos que el valor de la media cambia significativamente con tan sólo añadir 1 valor extremo.

### 2.2. La mediana

Si tenemos variables numéricas pero estas no tienen una distribución normal o tienen valores extremos, resulta más adecuado usar la mediana:

> La mediana es simplemente el valor medio de un conjunto de datos cuando estos están ordenados de menor a mayor

Ejemplo:

In [5]:
import numpy as np

arreglo1 = np.array([5,3,8,9,4])
print(f'Arreglo original: {arreglo1}')
print(f'Arreglo ordenado: {np.sort(arreglo1)}')
print(f'Mediana: {np.median(arreglo1)}')

Arreglo original: [5 3 8 9 4]
Arreglo ordenado: [3 4 5 8 9]
Mediana: 5.0


En el caso anterior, el arreglo original contiene un número impar de datos (5). Por tanto, después de ordenar el arreglo la mediana será el dato del medio (ubicado en la tercera posición).

Cuando el arreglo original contiene un número par de datos la mediana será un valor que no se encuentra en el arreglo:

In [6]:
arreglo2 = np.array([2, 9, 8, 16, 25, 1])
print(f'Arreglo original: {arreglo2}')
print(f'Arreglo ordenado: {np.sort(arreglo2)}')
print(f'Mediana: {np.median(arreglo2)}')

Arreglo original: [ 2  9  8 16 25  1]
Arreglo ordenado: [ 1  2  8  9 16 25]
Mediana: 8.5


Vemos que en este caso:

- El arreglo tiene 6 datos
- Por tanto no existe un dato intermedio. En su lugar los valor 8 y 9 son los valores intermedios
- La mediana es simplemente el valor medio entre estos dos valores intermedios: $mediana = \frac{8+9}{2} = 8.5$

La ventaja de la mediana, frente al uso de la media, es que es menos sensible a valores extremos.

Por ejemplo, agreguemos un valor extremo al arreglo anterior y recalculemos su mediana:

In [7]:
arreglo3 = np.array([2, 9, 8, 16, 25, 1, 800])
print(f'Arreglo original: {arreglo3}')
print(f'Arreglo ordenado: {np.sort(arreglo3)}')
print(f'Mediana: {np.median(arreglo3)}')

Arreglo original: [  2   9   8  16  25   1 800]
Arreglo ordenado: [  1   2   8   9  16  25 800]
Mediana: 9.0


Vemos que a pesar de tener un valor extremo muy grande (800) la mediana cambia tan sólo de 8.5 a 9.0.

### 2.3. La moda

Se usa como medida de tendencia central para datos categóricos:

> La moda es simplemente el valor más frecuente en la variable categórica

Por ejemplo, supongamos que tenemos la siguiente variable categórica:

In [8]:
arreglo4 = np.array(['A', 'B', 'A', 'C', 'B', 'A', 'A'])
print(arreglo4)

['A' 'B' 'A' 'C' 'B' 'A' 'A']


En este caso tenemos el siguiente conteo de frecuencias:

- `A`: 4
- `B`: 2
- `C`: 1

Por tanto, decimos que la moda es el valor `A`.

## Ejemplo práctico

Pongamos en práctica todos estos conceptos. Leamos el dataset `tasas_crimenes.csv` que contiene la tasa de asesinatos (por cada 100.000 habitantes) en diferentes estados de Estados Unidos.

Intentemos responder a estas preguntas:

1. ¿Cuáles son la media y la mediana de la tasa de asesinatos?
2. ¿Cuáles son la media y la mediana del número total de asesinatos?

Comencemos leyendo el set de datos:

In [9]:
# Leer dataset
import pandas as pd

ruta = '/Users/miguel/Library/CloudStorage/GoogleDrive-miguel@codificandobits.com/My Drive/02-CODIFICANDOBITS.COM/04-Academia/01-Cursos/21-2024-03-Estadística-Descriptiva/datos/'

df = pd.read_csv(ruta + 'tasas-crimenes.csv')
df

,State,Population,Murder.Rate,Abbreviation
0,Alabama,4779736,5.7,AL
1,Alaska,710231,5.6,AK
2,Arizona,6392017,4.7,AZ
3,Arkansas,2915918,5.6,AR
4,California,37253956,4.4,CA
5,Colorado,5029196,2.8,CO
6,Connecticut,3574097,2.4,CT
7,Delaware,897934,5.8,DE
8,Florida,18801310,5.8,FL
9,Georgia,9687653,5.7,GA


Para calcular la media y la mediana de la tasa de asesinatos simplemente podemos usar los métodos `mean()` y `median()` respectivamente:

In [10]:
print(f"Promedio tasa de asesinatos: {df['Murder.Rate'].mean()}")
print(f"Mediana tasa de asesinatos: {df['Murder.Rate'].median()}")

Promedio tasa de asesinatos: 4.066
Mediana tasa de asesinatos: 4.0


Vemos que tanto el promedio como la mediana tienen valores prácticamente iguales. Esto nos indica que los datos tienen una distribución simétrica (es decir a lado y lado de la media/mediana tenemos más o menos la misma distribución de datos).

Para calcular la media y la mediana del **número total** de asesinatos, primero tenemos que calcular el número total de asesinatos por cada ciudad:

*Número total de asesinatos = Población x Tasa de asesinatos / 100.000*

debido precisamente a que la tasa de asesinatos está dada por cada 100.000 habitantes:

In [11]:
df['Nro_asesinatos'] = (1/100000)*df['Murder.Rate']*df['Population']
df

,State,Population,Murder.Rate,Abbreviation,Nro_asesinatos
0,Alabama,4779736,5.7,AL,272.444952
1,Alaska,710231,5.6,AK,39.772936
2,Arizona,6392017,4.7,AZ,300.424799
3,Arkansas,2915918,5.6,AR,163.291408
4,California,37253956,4.4,CA,1639.174064
5,Colorado,5029196,2.8,CO,140.817488
6,Connecticut,3574097,2.4,CT,85.778328
7,Delaware,897934,5.8,DE,52.080172
8,Florida,18801310,5.8,FL,1090.475980
9,Georgia,9687653,5.7,GA,552.196221


Y con estos valores totales por cada ciudad, simplemente calculamos la media y la mediana de esta nueva columna:

In [12]:
print(f"Promedio de asesinatos: {df['Nro_asesinatos'].mean()}")
print(f"Mediana de asesinatos: {df['Nro_asesinatos'].median()}")

Promedio de asesinatos: 273.99124876
Mediana de asesinatos: 162.662234


Acá sí hay una diferencia notable entre la media y la mediana (274 vs. 163 asesinatos). Esto se debe a que las poblaciones se encuentran dentro de un rango amplio de valores, lo que al momento del cálculo afecta el número total de asesinatos:

In [13]:
df.describe()

,Population,Murder.Rate,Nro_asesinatos
count,5.000000e+01,50.000000,50.000000
mean,6.162876e+06,4.066000,273.991249
std,6.848235e+06,1.915736,323.216195
min,5.636260e+05,0.900000,10.011856
25%,1.833004e+06,2.425000,54.193103
50%,4.436370e+06,4.000000,162.662234
75%,6.680312e+06,5.550000,359.342657
max,3.725396e+07,10.300000,1639.174064


Vemos que efectivamente el número de asesinatos oscila entre 10 y 1639, un rango muy amplio que hace que muy probablemente haya *outliers* (ciudades con un número o muy bajo o muy alto de asesinatos).

Por tanto, en este caso del número de asesinatos, una medida más fiable es la mediana (163 asesinatos).

## 4. Resumen

|  | Variable numérica | Variable categórica | ¿Cuándo usar? | Limitaciones |
|:---|:---|:---|:---|:---|
| Media | ✔︎ |  | Distribución normal y no hay presencia de *outliers* | Sensible a los *outliers* |
| Mediana | ✔︎ |  | Distribución no normal y cuando hay valores extremos  |  |
| Moda |  | ✔︎ |  |  |